main/glossary/README.md

```diff
@@ -67,151 +104,379 @@ synchronously compared for structural equivalence.
+what the seat's current [allocation](#allocation) is. 
+
+If there was a previous reallocation, the payout is different than what the user escrowed. If there is no reallocation
+before the seat exits, the payout is the same as what they escrowed.
+
+## Petname
+
+Petnames are your personal names for objects. No one else can see or modify a petname without your permission. 
+Think of them as similar to a phone's contacts list. The actual phone number is what a phone uses to call 
+someone, but to more easily tell who a number is associated with, it's assigned a petname, such 
+as Mom, Grandpa, Kate S., etc. In the Agoric platform, petnames are used in [wallets](#wallet).
+
+## Presence 
+A local version of a remote object that serves as the remote object's proxy. 
 If `obj` is a presence of a remote object, you can send messages to the remote object by using `E()` on `obj`. 
-See the [JavaScript Distributed Programming Guide](/distributed-programming.md) for more information. 
+For more information, see the [JavaScript Distributed Programming Guide](/distributed-programming.md). 
+
+## Proposal
+
+Proposals are records with `give`, `want`, and `exit` keys. [Offers](#offer) must include a proposal, which states
+what asset you want, what asset you will give for it, and how/when the offer maker can cancel the offer
+(see [Exit Rule](#exit-rule) for details on the last). For example:
+```
+const myProposal = harden({
+  give: { Asset: quatloosAmountMath.make(4)},
+  want: { Price: moolaAmountMath.make(15) },
+  exit: { 'onDemand'
+})
+```
+`give` and `want` use [keywords](#keywords) defined by the contract. Each specifies via an [amount](#amounts), a description of what
+asset they are willing to give/want to get, and how much of it. 
+
+## Purse 
+Purses hold [amounts](#amounts) of a certain [mint](#mint) issued assets. Specifically amounts that are _stationary_.
+Purses can transfer part of their held balance to a [payment](#payment), which is usually used to transfer value. 
+A purse's contents are all of the same [brand](#brand).
 
-## Purse
-Purses hold [amounts](#amounts) of certain [mint](#mint) issued assets. Specifically amounts that are _stationary_. Purses can transfer part of their held balance to a [payment](#payment), which is usually used to transfer value. A purse's contents are all of the same [brand](#brand).
+For more information, see the [ERTP Guide's Purses section](/ertp/guide/purses-and-payments.md#purses-and-payments) and the
+[ERTP API's Purses section](/ertp/api/purse.md).
 
-See the [ERTP Guide's Purses section](/ertp/guide/purses-and-payments.md#purses) and the
-[ERTP API's Purses section](/ertp/api/purse.md)
-for more information.
+## Quatloos
+An imaginary currency Agoric documentation uses in examples. For its origins, see the Wikipedia entry for the Star Trek 
+episode [The Gamesters of Triskelion](https://en.wikipedia.org/wiki/The_Gamesters_of_Triskelion).
+
+## Reallocate/Reallocation
```

Comment:
For the purposes of the link, let's only use one of these. If we need another entry for the other title, we can point them to this one.